server/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -278,10 +278,10 @@ private GetResult innerGetLoadFromStoredFields(String id, String[] storedFields,
                 documentFields = new HashMap<>();
                 metadataFields = new HashMap<>();
                 for (Map.Entry<String, List<Object>> entry : fieldVisitor.fields().entrySet()) {
-                    if (MapperService.isMetadataField(entry.getKey())) {
-                        metadataFields.put(entry.getKey(), new DocumentField(entry.getKey(), entry.getValue()));
+                    if (mapperService.isMetadataField(entry.getKey())) {
+                        metadataFields.put(entry.getKey(), new DocumentField(entry.getKey(), entry.getValue(), true));
```

Comment:
Again, curious why we need the additional flag on DocumentField, if the maps are separate here (and stay so as far as I can see in GetResult). I'm probably missing the point in the code where this isn't easily reconstructable.